tests/blake2b.cc

```diff
@@ -148,67 +87,150 @@ read_official_json()
-                           input,
-                           test_case.key.size(),
-                           key,
-                           test_case.out_len,
-                           digest);
-  EXPECT_TRUE(test);
-
-  if (test_case.key.size() == 0) {
-    bytes digest_evercrypt(test_case.out_len, 0);
-    uint8_t* input = const_cast<uint8_t*>(test_case.input.data());
-    EverCrypt_Hash_hash(EverCrypt_Hash_Blake2B_s,
-                        digest_evercrypt.data(),
-                        input,
-                        test_case.input.size());
-    EXPECT_EQ(digest_evercrypt, test_case.digest);
+  auto test = GetParam();
+
+  {
+    bytes got_digest(test.out_len);
+
+    Hacl_Blake2b_32_blake2b(test.out_len,
+                            got_digest.data(),
+                            test.input.size(),
+                            test.input.data(),
+                            test.key.size(),
+                            test.key.data());
+
+    bool outcome =
+      compare_and_print(test.out_len, got_digest.data(), test.digest.data());
+
+    EXPECT_TRUE(outcome);
+  }
+
+  // EverCrypt
+  {
+    EverCrypt_AutoConfig2_init();
+
+    if (test.key.size() == 0) {
+      bytes got_digest(test.out_len, 0);
+
+      EverCrypt_Hash_hash(EverCrypt_Hash_Blake2B_s,
+                          got_digest.data(),
+                          test.input.data(),
+                          test.input.size());
+
+      EXPECT_EQ(got_digest, test.digest);
+    }
   }
 }
 
-TEST_P(Blake2bKAT, TryKATStreaming)
+class Blake2bStreaming : public ::testing::TestWithParam<TestCase>
+{};
+
+TEST_P(Blake2bStreaming, KAT)
 {
-//  hacl_init_cpu_features();
-//  if (hacl_vec256_support()) {
-//    printf("Support");
-//  } else {
-//    printf("No support.");
-//  }
-}
+  auto test_case = GetParam();
+
+  {
+    // Skip tests with key.
+    if (test_case.key.size() != 0) {
+      return;
+    }
+
+    bytes got_digest(64);
+
+    // Init
+    Hacl_Streaming_Blake2_blake2b_32_state_s* state =
+      Hacl_Streaming_Blake2_blake2b_32_no_key_create_in();
+    Hacl_Streaming_Blake2_blake2b_32_no_key_init(state);
 
-INSTANTIATE_TEST_SUITE_P(Kat, Blake2bKAT, ::testing::ValuesIn(read_json()));
+    // Update
+    Hacl_Streaming_Blake2_blake2b_32_no_key_update(
+      state, test_case.input.data(), test_case.input.size());
 
-INSTANTIATE_TEST_SUITE_P(OfficialKat,
-                         Blake2bKAT,
-                         ::testing::ValuesIn(read_official_json()));
+    // Finish
+    Hacl_Streaming_Blake2_blake2b_32_no_key_finish(state, got_digest.data());
+    Hacl_Streaming_Blake2_blake2b_32_no_key_free(state);
+
+    EXPECT_EQ(test_case.digest, got_digest);
+  }
 
-INSTANTIATE_TEST_SUITE_P(OfficialKatStreaming,
-                         Blake2bKAT,
-                         ::testing::ValuesIn(read_official_json()));
+#ifdef HACL_CAN_COMPILE_VEC256
+  {
+    hacl_init_cpu_features();
+
+    if (hacl_vec256_support()) {
+      //    bytes got_hash(64);
```

Comment:
Please add a todo with a link to the upstream issue so we know why it's commented out.